packages/block-library/src/blocks/sidecar/scss/_block-content-alignments.scss

```diff
@@ -0,0 +1,336 @@
+// we are going to consider
+// user intention of pulling the block and pull
+// the block on the right side.
+//
+// The same logic is applied for alignright and pull-right.
+.alignleft,
+.pull-left,
+[data-novablocks-alignment='pull-left'] {
+    
+  --block-content-start: ws;
+  --block-content-end: gs;
+
+  &.stop-left {
+  
+    --block-content-start: ge;
+    --block-content-end: we;
+  
+    + *:not([class*='align']) {
+      --block-content-end: ce;
+    }
+  }
+}
+
+.alignright,
+.pull-right,
+[data-novablocks-alignment='pull-right'] {
+  
+  --block-content-start: ge;
+  --block-content-end: we;
+
+  &.stop-right {
+    --block-content-start: ws;
+    --block-content-end: gs;
+  
+    + *:not([class*='align']) {
+      --block-content-start: cs;
+    }
+  }
+}
+
+// To avoid unneeded white space,
+// aligned and pulled items will
+// span on 3 rows.
+.alignright,
+.alignleft,
+.pull-right,
+.pull-left {
+  grid-row-end: span 5;
+}
+
+// ALIGNMENTS
+
+.novablocks-content {
+  grid-column: var(--block-content-start)/var(--block-content-end);
+  position: relative;
+}
+
+// RIGHT SIDEBAR
+.novablocks-sidecar {
+  
+  .novablocks-content:not(:only-child) {
+  
+    .pull-left,
+    .pull-right,
+    .alignleft,
+    .alignright {
+      @include grid-item-position('right');
+      
+      &[class*='stop'] {
+        --block-content-start: var(--block-limit-left);
+        --block-content-end: ce;
+  
+        + *:not([class*='align']) {
+          --block-content-start: var(--block-limit-left);
+        }
+      }
+    }
+  }
+}
+
+// LEFT SIDEBAR
+.novablocks-sidecar--sidebar-left {
+  
+  .novablocks-content:not(:only-child)[class] {
+  
+    .pull-left,
+    .pull-right,
+    .alignleft,
+    .alignright {
+      @include grid-item-position('left');
+  
+      + *:not([class*='align']) {
+        --block-content-start: cs;
+        --block-content-end: var(--block-limit-right);
+      }
+      
+      &[class*='stop'] {
+        --block-content-start: cs;
+        --block-content-end: var(--block-limit-right);
+      }
+    }
+  }
+}
+
+// TWO SIDEBARS
+.novablocks-sidecar--complex {
+  
+  .novablocks-content:not(:only-child)[class] {
+  
+    .pull-left {
+  
+      @include grid-item-position('left');
+  
+      // There is not space on the left side,
+      // so we are going to treat pull left as pull right.
+      &.stop-left {
+        @include grid-item-position('right');
+      }
+    }
+  
+    .pull-right {
+  
+      @include grid-item-position('right');
+    
+      // There is not space on the left side,
+      // so we are going to treat pull left as pull right.
+      &.stop-right {
+        @include grid-item-position('left');
+      }
+    }
+    
+    .alignleft {
+  
+      @include grid-item-position('left');
+      
+      &[class*='stop-left'] {
+        @include grid-item-position('right');
+      }
+      
+      &[class*='stop-right'] {
+        @include grid-item-position('left');
+      }
+    }
```

Comment:
This selectors could be grouped and the `grid-item-position` mixin be used only twice. This isn't very costly since we're talking about only two lines of code for each mixin, so if you feels it helps legitibiliy we can keep them this way.